idflakies-core/src/main/java/edu/illinois/cs/dt/tools/detection/detectors/DetectorFactory.java

```diff
@@ -24,8 +36,16 @@ public static Detector makeDetector(final InstrumentingSmartRunner runner, final
             return new OriginalDetector(runner, baseDir, rounds, tests);
         } else if (detectorType().equals("smart-shuffle")) {
             return new SmartShuffleDetector(runner, baseDir, rounds, tests, detectorType());
+        }  else if (detectorType().equals("tuscan")){
+            int n = getClassesSize(tests);
+            if(n == 3 || n == 5) {
+                return new TuscanDetector(runner, baseDir, rounds > (n+1) ? (n+1) : rounds, detectorType(), tests);
```

Comment:
Does it have to distinguish number of rounds here upon creation? Can it be handled within the detector itself (so the additional code to be modified in this existing class can be simplified)?

Reply:
I have moved them into TuscanDetector.java and it worked same as before.